src/librbd/mirror/snapshot/CreatePrimaryRequest.cc

```diff
@@ -207,6 +207,10 @@ void CreatePrimaryRequest<I>::unlink_peer() {
       }
       count++;
       if (count == 3) {
+        if (info->mirror_peer_uuids.count(peer) == 0
+            && info->mirror_peer_uuids.size() != 0) {
+          break;
+        }
```

Comment:
@MrFreezeex I am not sure it is right. Why do you just break without setting `unlink_snap_id` and then `snap_id` and `peer_uuid`?

Looking at it, wouldn't we just need to add this check before `count++` line and just continue the loop if it is false?

Reply:
Hi sorry, I was waiting for the test to pass to comment. It executes this `break` statement only in the inner loop so this should work as expected. I didn't want to change the counter behavior, but what you are suggesting should be fine too. I'm changing it now!